### ISSUE TITLE
Docker 1.7 breaking API change Host -> HostConfig

### ISSUE THREAD

**Issue by tnolet** [docker/docker#14281]
Hi,

Description of problem:

we noticed our application started throwing errors because some options in the `docker inspect` output were moved from the `Config` object to the `HostConfig` object, `CpusSet` and `CpuShare` for example.
I can imagine  A LOT of Docker users have applications that use some form of static typing to talk to the Docker daemon. This will now break in many cases.

This is a major concern as we have noticed these types of breaking changes from 1.5 -> 1.6 and now to 1.7.

`docker version`: 
```
Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64
```

`docker info`:
```
Containers: 1
Images: 7
Storage Driver: devicemapper
 Pool Name: docker-8:1-34206275-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 321.5 MB
 Data Space Total: 107.4 GB
 Data Space Available: 9.121 GB
 Metadata Space Used: 938 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 1
Total Memory: 3.614 GiB
Name: instance-1
ID: JCTD:QNL3:KXDV:PM7K:IWWW:TPLU:OKRG:UZZG:K3JH:43O5:VXYQ:2MWN
```

`uname -a`:
```
Linux instance-1 3.10.0-229.4.2.el7.x86_64 #1 SMP Wed May 13 10:06:09 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Environment details (AWS, VirtualBox, physical, etc.): GCE Virtual machine.

How reproducible:

Run any container and then do a `docker inspect` with Docker 1.7. Compare the output to exactly the same procedure running Docker 1.6.x

Actual Results:

The output from Docker inspect now lists the `CpuSet`, `CpuShares` and other fields under the 'HostConfig` Map. See the shortened output.

```json
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LxcConf": [],
            "Memory": 0,
            "MemorySwap": 0,
            "CpuShares": 0,
            "CpuPeriod": 0
         }
```

**Comment by unclejack** [comment]
@tnolet I understand that this is very frustrating, that running into this problem can be very annoying and I'm sorry you've had problems with this. However, this change had to be made because that's where those options need to be. This isn't a bug or a regression, it's merely about having moved some options from one place to where they need to be. The resource limits are supposed to be in HostConfig, not in Config.

Since this isn't a bug and the changes will not be reverted because this is the right thing to do, I'm going to close this issue. If you happen to come across this issue and want to discuss it further, please don't open another one and just comment on this one.

**Comment by tnolet** [comment]
@unclejack I didn't file it as a bug, more of a notice that these are unexpected and undocumented breaking changes. I couldn't find any mention of this in the release notes or changelog. A simple heads up would certainly help. Now its just "Boom", nothing works anymore.

**Comment by hqhq** [comment]
@tnolet We usually consider backward compatibility as very high priority job, when I made this resource config change, we do have backward compatibility, but it's a bit annoying for the code and that part of code was removed in a code refactor.

However it should be in CHANGELOG.md, but that's not hard requirement and developers usually forget to record these information. We should do it better in the future.
